screen2.0/src/common/components/mainsearch/GeneAutocomplete.tsx

```diff
@@ -0,0 +1,192 @@
+import React, { useState, useEffect, useCallback } from "react"
+import Box from "@mui/material/Box"
+import TextField from "@mui/material/TextField"
+import Autocomplete from "@mui/material/Autocomplete"
+import Grid from "@mui/material/Grid"
+import Typography from "@mui/material/Typography"
+import { debounce } from "@mui/material/utils"
+import { useRouter } from "next/navigation"
+export type QueryResponse = [number, string[], any, [string, string, string, string, string, string][], string[]]
+
+const GENE_AUTOCOMPLETE_QUERY = `
```

Comment:
Queries.ts file? Or is it best to define queries where they're used if they're only used once?